Stork config leaks between instances in a continuous testing run

### Describe the bug

This reproduces with 3.20.0, so it's not caused by the test classloading rewrite. :) 
It also reproduces on main, and https://github.com/quarkusio/quarkus/pull/47180 exposes it for the `mvn verify` case, which is of course worse.

### Steps to reproduce 

- check out superheroes
- `cd rest-fights`
- Edit `HeroesVillainsNarrationWiremockServerResource` and delete the `quarkus.stork.hero-service.service-discovery.address-list` config override, so that the application is always reading config from `application.properties`. I don't think this step is strictly necessary, but it makes reproducing and searching easier, because the 'wrong' port is a predictable one.

- Now if you run `quarkus test `you'll get 4-12 failures. on 3.21.0 you will see 8 pre-existing failures, caused by #47177, main will be a bit less red. But in either case there will be extra, expected, failures caused by the fact that you're trying to connect to an endpoint you haven't mocked. You can search the logs for `8083` for those. 

- Then edit `application.properties` and change 8083 to 1234. If you look at the logs, you'll get messages saying "I can't connect to port 1234" (expected), but some of the messages will still reference port 8083. Nothing should know about 8083, but something does.

To get a smaller reproducer, you can delete everything except `HeroClientTest`.

### Stack traces

```
09:29:45 ERROR [io.qu.test] (Test runner thread) ==================== TEST REPORT #3 ==================== [Error Occurred After Shutdown]
09:29:45 ERROR [io.qu.test] (Test runner thread) Test HeroClientTests#helloHeroes() failed 
 [Error Occurred After Shutdown]: java.lang.AssertionError: Expected an item event but got a failure: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:8083
        at io.smallrye.mutiny.helpers.test.UniAssertSubscriber.awaitItem(UniAssertSubscriber.java:165)
        at io.quarkus.sample.superheroes.fight.client.HeroClientTests.helloHeroes(HeroClientTests.java:187)


09:29:45 ERROR [io.qu.test] (Test runner thread) Test HeroClientTests#doesntRecoverFrom500() failed 
 [Error Occurred After Shutdown]: java.lang.AssertionError: 
Received a failure event, but expecting:
  <io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:1234>
to be an instance of:
  <class jakarta.ws.rs.WebApplicationException>
but was:
  <io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:1234
Caused by: java.net.ConnectException: Connection refused
        at java.base/sun.nio.ch.Net.pollConnect(Native Method)
```

### Why do I think stork is involved?

If you stop the application and change 

```
quarkus.rest-client.hero-client.url=stork://hero-service
```

to 

```
quarkus.rest-client.hero-client.url=http://localhost:4321
```

and then run `quarkus test`, you get expected failures references 4321. If you then change 4321 to 8888 in `application.properties` and save, there are no more references to 4321 in the new logging.

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -59,7 +59,7 @@
         <smallrye-reactive-types-converter.version>3.0.1</smallrye-reactive-types-converter.version>
         <smallrye-mutiny-vertx-binding.version>3.18.1</smallrye-mutiny-vertx-binding.version>
         <smallrye-reactive-messaging.version>4.27.0</smallrye-reactive-messaging.version>
-        <smallrye-stork.version>2.7.2</smallrye-stork.version>
+        <smallrye-stork.version>2.7.3</smallrye-stork.version>
         <jakarta.activation.version>2.1.3</jakarta.activation.version>
         <jakarta.annotation-api.version>3.0.0</jakarta.annotation-api.version>
         <jakarta.authentication-api>3.0.0</jakarta.authentication-api>

--- a/independent-projects/resteasy-reactive/pom.xml
+++ b/independent-projects/resteasy-reactive/pom.xml
@@ -62,7 +62,7 @@
         <rest-assured.version>5.5.1</rest-assured.version>
         <commons-logging-jboss-logging.version>1.0.0.Final</commons-logging-jboss-logging.version>
         <jackson-bom.version>2.18.3</jackson-bom.version>
-        <smallrye-stork.version>2.7.2</smallrye-stork.version>
+        <smallrye-stork.version>2.7.3</smallrye-stork.version>
         <jakarta.validation-api.version>3.0.2</jakarta.validation-api.version>
         <yasson.version>3.0.4</yasson.version>
         <jakarta.json.bind-api.version>3.0.1</jakarta.json.bind-api.version>

